app/src/main/res/layout/fragment_trip_detail.xml

```diff
@@ -87,24 +89,105 @@
 				android:singleLine="true"
 				android:textColor="?android:textColorPrimary"
 				app:layout_constraintBottom_toBottomOf="parent"
-				app:layout_constraintEnd_toStartOf="@+id/duration"
+				app:layout_constraintEnd_toStartOf="@+id/durationPrice"
 				app:layout_constraintStart_toEndOf="@+id/toTime"
 				app:layout_constraintTop_toBottomOf="@+id/from"
 				tools:text="To Station"/>
 
+			<LinearLayout
+				android:id="@+id/durationPrice"
+				android:layout_width="wrap_content"
+				android:layout_height="match_parent"
+				android:orientation="vertical"
+				android:layout_marginTop="8dp"
+				android:layout_marginEnd="8dp"
+				android:layout_marginRight="8dp"
+				android:layout_marginBottom="8dp"
+				android:gravity="center_vertical"
+				app:layout_constraintEnd_toEndOf="parent"
+				app:layout_constraintTop_toTopOf="parent"
+				app:layout_constraintBottom_toBottomOf="parent">
+
+				<TextView
+					android:id="@+id/duration"
+					android:layout_width="wrap_content"
+					android:layout_height="wrap_content"
+					android:layout_gravity="end"
+					android:textColor="?android:textColorSecondary"
+					tools:text="0:42"/>
+
+				<TextView
+					android:id="@+id/price"
+					android:layout_width="wrap_content"
+					android:layout_height="wrap_content"
+					android:layout_gravity="end"
+					android:textColor="?android:textColorSecondary"
+					tools:visibility="gone"
+					tools:text="1,50 €"/>
+
+			</LinearLayout>
+
+		</androidx.constraintlayout.widget.ConstraintLayout>
+
+		<androidx.constraintlayout.widget.ConstraintLayout
+			android:id="@+id/topBar"
+			android:layout_width="match_parent"
+			android:layout_height="wrap_content"
+			android:layout_marginTop="0dp"
+			android:background="?android:colorBackground"
+			tools:visibility="visible">
```

Comment:
Why a new ConstraintLayout?

Reply:
Because the two ConstraintLayouts are not shown at the same time, either the `bottomBar` or the `topBar` is.

Reply:
Sorry can you explain a bit more?

Reply:
Sure, `bottomBar` is the what's visible if the map is spanning almost all of the screen and the start and end points of the journey are just shown in the bottom line (with red background). Whereas `topBar` is the newly added line above the journey details as proposed in https://github.com/grote/Transportr/issues/231#issuecomment-355644068.

This means both are never visible at the same time and are thus completely independent one from another. In [TripDetailFragment.kt](https://github.com/grote/Transportr/pull/627/files#diff-112f199a9477cb8f84092554d964f660R169), the visibility is switched for both ConstraintLayouts whenever the sheet state has changed. That's why I went with having two separate layout "groups" which can be easily shown or hidden altogether.